addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/config/NestDeviceConfiguration.java

```diff
@@ -8,14 +8,17 @@
  */
 package org.openhab.binding.nest.internal.config;
 
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
 /**
  * The configuration for Nest devices.
  *
  * @author Wouter Born - Initial contribution
  * @author Wouter Born - Add device configuration to allow file based configuration
  */
+@NonNullByDefault
 public class NestDeviceConfiguration {
     public static final String DEVICE_ID = "deviceId";
     /** Device ID which can be retrieved with the Nest API. */
-    public String deviceId;
+    public String deviceId = "";
```

Comment:
Will / can it never be set to `null`?

Reply:
It's a required configuration parameter see the Nest [config.xml](https://github.com/wborn/openhab2-addons/blob/d3e2a278b8b101e4b98133b20ed088dac840321a/addons/binding/org.openhab.binding.nest/ESH-INF/config/config.xml#L48).